cmd/ipfs-cluster-service/main.go

```diff
@@ -632,3 +641,36 @@ func getStateManager() cmdutils.StateManager {
 	checkErr("creating state manager", err)
 	return mgr
 }
+
+func assignRandomPorts(multiAddrs []*ma.Multiaddr) {
+	var prev string
+	for _, m := range multiAddrs {
+		components := []ma.Multiaddr{}
+		ma.ForEach(*m,
+			func(c ma.Component) bool {
+				code := c.Protocol().Code
+				var port int
+				if code == ma.P_TCP || code == ma.P_UDP {
```

Comment:
too many levels of indentation here

Reply:
edited code contains fewer blocks now, thus fewer levels of indentations